api/app/routers/hfi_calc.py

```diff
@@ -191,12 +191,35 @@ async def set_planning_area_station_fuel_type(
     token=Depends(authentication_required)  # pylint: disable=unused-argument
 ):
     """ Set the fuel type for a station in a planning area. """
-    # TODO: stub - implement!
     logger.info("/fire_centre/%s/%s/%s/planning_area/%s/station/%s/fuel_type/%s",
                 fire_centre_id, start_date, end_date,
                 planning_area_id, station_code, fuel_type_id)
     response.headers["Cache-Control"] = no_cache
-    raise NotImplementedError('This function is not implemented yet.')
+
+    with get_read_session_scope() as session:
+        # We get an existing request object (it will load from the DB or create it
+        # from scratch if it doesn't exist).
+        request, _, fire_centre_fire_start_ranges = get_prepared_request(
+            session,
+            fire_centre_id,
+            DateRange(start_date=start_date,
+                      end_date=end_date))
+
+        # Validate the fuel type id.
+        fuel_type = get_fuel_type_by_id(session, fuel_type_id)
+        if fuel_type is None:
+            raise HTTPException(status_code=500, detail="Fuel type not found")
```

Comment:
Strange that codecov is complaning about this when you do have a test for this case...

Reply:
Good catch - test was looking for 500, and yes - it was throwing a 500 - but not for the right reason. Bug in the unit test mock function.